src/observers/stores/datasets.py

```diff
@@ -24,21 +24,15 @@
 def push_to_hub(self):
     """Push pending changes to the Hugging Face Hub"""
     with self.lock:
-        records = [
-            json.loads(line)
-            for json_file in Path(self.folder_path).rglob("*.json")
-            for line in open(json_file)
-        ]
-
-        for record in records:
-            if "image" in record and "path" in record["image"]:
-                image_path = Path(self.folder_path) / record["image"]["path"]
-                record["image"] = str(image_path)
-
-        dataset = Dataset.from_list(records).cast_column("image", DatasetImage())
-        dataset.push_to_hub(
-            repo_id=self.repo_id, token=self.token, private=self.private
-        )
+        for json_file in Path(self.folder_path).rglob("*.json"):
+            upload_file(
+                path_or_fileobj=str(json_file),
+                path_in_repo=json_file.name,
+                repo_id=self.repo_id,
+                repo_type="dataset",
+                token=self.token,
+                commit_message=f"Upload {json_file.name}",
+            )
 
```

Comment:
Will have to think about how we can preserve image like we do before but also support uploading files. We need to upload JSON

Reply:
@cfahlgren1 https://huggingface.co/datasets/davidberenstein1957/python-math-assistant/tree/main/data something like this should work.

Reply:
I will write some code for that now.